src/dsp_tools/utils/shared.py

```diff
@@ -98,12 +99,12 @@ def try_network_action(
     raise BaseError(failure_msg)
 
 
-def validate_xml_against_schema(input_file: str) -> bool:
+def validate_xml_against_schema(xml_file_as_path_or_parsed: Union[str, etree.ElementTree]) -> bool:  # type: ignore
```

Comment:
why does this need the type ignore? Because of the ElementTree? If I recall correctly, the actual type returned by `etree.ElementTree` is `etree._ElementTree` or something like that. It is definitely possible to use types correctly with lxml, and if type hints are not used correctly, they are better not used at all, otherwise it's only confusing for programmers, users and tools.

And content-wise: Does it really make sense to introduce this change? It violates the principle of single responsibility and makes both the function signature and the function body more complex. I think it would be much cleaner to have two different functions to take parsed xml or the path, and the path-function would read the file, parse it and then call the other function.

Reply:
I always struggle with type annotations of lxml, even though I have types-lxml installed. types-lxml doesn't even seem to help me. I struggle that much that I seriously consider switching to the stdlib https://docs.python.org/3/library/xml.html. 

`etree._ElementTree` doesn't work: `Missing type parameters for generic type "_ElementTree"  [type-arg]mypy(error)`

So I tried `etree._ElementTree[Any]`, and this seems to work. At least, the tests run without problem, so it's probably a good solution, even though I'm a bit sceptic about the `[Any]`

Reply:
Have you tried `etree._ElementTree[etree._Element]`?  
(and maybe for the sanity of your eyes, define a type alias for that...)

And yes, if lxml is hampering you, it would make sense to switch. I'm not sure though, if you can switch entirely - I don't think the standard library includes schema validation. You'd have to check.